### PR TITLE
download: unify handling, refuse while under test

### LINKED ISSUE #6113: Error parsing netrc file: permission denied
minikube start
😄  minikube v1.6.1 on Ubuntu 16.04
✨  Automatically selected the 'virtualbox' driver (alternates: [none])
💿  Downloading VM boot image ...

💣  Failed to cache ISO: https://storage.googleapis.com/minikube/iso/minikube-v1.6.0.iso: invalid checksum: Error downloading checksum file: Error parsing netrc file at "/home/miracle/.netrc": open /home/miracle/.netrc: permission denied

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:

### PR DESCRIPTION
**Pull request by tstromberg**
* Refuse to download actual files if we detect that we are in a test
* Add a mechanism for tests to state that they wish to have mock downloads
* Unify atomic download code among functions
* Remove `binary_test`, which was an integration test masquerading as a unit test
* Disable netrc, as we don't use it, and it has caused users permission problems (#6113  #5643)

Obsoletes #6907 #7307
Fixes #6113  #5643

### CODE REVIEW

File: pkg/minikube/download/download.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	tmpDst := dst + ".download"
+	client := &getter.Client{
+		Src:     src,
+		Dst:     tmpDst,
+		Dir:     false,
+		Mode:    getter.ClientModeFile,
+		Options: []getter.ClientOption{getter.WithProgress(DefaultProgressBar)},
+		Getters: map[string]getter.Getter{
+			"file":  &getter.FileGetter{Copy: false},
+			"http":  &getter.HttpGetter{Netrc: false},
```
Comment (medyagh) [inline pkg/minikube/download/download.go:47]: please add a reason in PR description why you disable Netrc

Comment (tstromberg) [reply pkg/minikube/download/download.go:47]: clarified. Thanks!

**Comment by medyagh** [comment]
this test is scary
driver_install_or_update_test.go:89: Failed to update driver to 1.3.0. test: driver-without-version-support, got: download: unmocked download under test

**Pull request merged**